Add Sourcegraph Datadog Services Map Extension integration listing

Joelkw opened the pull request:
### What does this PR do?

This PR adds a tile-only listing for the Sourcegraph Datadog Services Map extension integration. The integration can also be found here: https://sourcegraph.com/extensions/sourcegraph/datadog-service-map . 

I've been working with Borja Burgos and Kai Xin Tai on the Datadog side (as well as Omri Sass and his team, who were very helpful), and we're ready to release and start promoting this. They're expecting this PR. 

### Motivation

This is a new feature that makes clever use of the new [Datadog Service Dependencies API](https://docs.datadoghq.com/api/latest/service-dependencies/) to let you navigate, read, and debug code based on knowing what services it calls or it is called by, with an option to take you directly to the Datadog Services Map for full context. 

### Review checklist

- [ ] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [ ] Feature or bugfix has tests
- [ ] Git history is clean
- [ ] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes

Anything else we should know when reviewing?

As noted, feel free to cross-check anything with Borja or Kai.

Joelkw commented on sourcegraph/manifest.json:
```diff
@@ -0,0 +1,33 @@
+      "metadata_path": "assets/service_checks.json"
+    }
+  }
+}
```
```suggestion
}

```

kayayarai commented on sourcegraph/README.md:
```diff
@@ -0,0 +1,69 @@
+
+## Overview
+
+![Datadog Services Map Screenshot](images/sourcegraph-datadog-services-map-extension.png)
```
```suggestion
![Datadog Services Map Screenshot](https://raw.githubusercontent.com/DataDog/integrations-extras/master/sourcegraph/images/sourcegraph-datadog-services-map-extension.png)
```

kayayarai commented on sourcegraph/README.md:
```diff
@@ -0,0 +1,69 @@
+
+![Datadog Services Map Screenshot](images/sourcegraph-datadog-services-map-extension.png)
+
+The Datadog Services Map Sourcegraph Extension brings the context of Datadog APM directly into your code on Sourcegraph and on your code host (via [the Sourcegraph Browser Extension](https://docs.sourcegraph.com/integration/browser_extension)). 
```
```suggestion
The Datadog Services Map Sourcegraph Extension brings the context of Datadog APM directly into your code on Sourcegraph and on your code host through [the Sourcegraph Browser Extension][1]. 
```
Links should be indexed and listed at the end of the file.

kayayarai commented on sourcegraph/README.md:
```diff
@@ -0,0 +1,69 @@
+If you're a an individual Sourcegraph user, email support@sourcegraph.com. 
+
+If you're a Sourcegraph customer, use your existing support slack channel or email contact.
+
```
```suggestion
[1]: https://docs.sourcegraph.com/integration/browser_extension
[2]: https://docs.datadoghq.com/tracing/visualization/services_map/
[3]: https://docs.datadoghq.com/help/
```

kayayarai commented on sourcegraph/README.md:
```diff
@@ -0,0 +1,69 @@
+
+The Datadog Services Map Sourcegraph Extension brings the context of Datadog APM directly into your code on Sourcegraph and on your code host (via [the Sourcegraph Browser Extension](https://docs.sourcegraph.com/integration/browser_extension)). 
+
+* See which services call or are called by other services 
```
```suggestion
* See which services call or are called by other services.
```

kayayarai commented on sourcegraph/README.md:
```diff
@@ -0,0 +1,69 @@
+The Datadog Services Map Sourcegraph Extension brings the context of Datadog APM directly into your code on Sourcegraph and on your code host (via [the Sourcegraph Browser Extension](https://docs.sourcegraph.com/integration/browser_extension)). 
+
+* See which services call or are called by other services 
+* Go directly to the [Datadog APM Services Map](https://docs.datadoghq.com/tracing/visualization/services_map/) for your service 
```
```suggestion
* Go directly to the [Datadog APM Services Map][2] for your service.
```

kayayarai commented on sourcegraph/README.md:
````diff
@@ -0,0 +1,69 @@
+"datadogServiceMap.corsAnywhereUrl": "CORS_ANYWHERE_URL",
+```
+
+> If you don't have a CORS anywhere URL, you can use the Sourcegraph demo: `https://sourcegraph-demo-cors-anywhere.herokuapp.com`
````
```suggestion
If you don't have a CORS anywhere URL, you can use the Sourcegraph demo: `https://sourcegraph-demo-cors-anywhere.herokuapp.com`
```

kayayarai commented on sourcegraph/README.md:
```diff
@@ -0,0 +1,69 @@
+This extension does not include any events.
+
+## Support
+
```
```suggestion

## Further reading

- [Use Datadog's Sourcegraph extension to navigate code and visualize service dependencies][4]
```

kayayarai commented on sourcegraph/README.md:
```diff
@@ -0,0 +1,69 @@
+Need help? Contact [Datadog support][3].
+[1]: https://docs.sourcegraph.com/integration/browser_extension
+[2]: https://docs.datadoghq.com/tracing/visualization/services_map/
+[3]: https://docs.datadoghq.com/help/
```
```suggestion
[3]: https://docs.datadoghq.com/help/
[4]: https://www.datadoghq.com/blog/sourcegraph-datadog-apm-integration/
```

kayayarai reviewed (approved):
One more suggested addition: a link to the new blog post!

Joelkw commented:
Hello @kayayarai ! Thank you for the great suggestions and help here. Is this okay to merge?

kayayarai commented on sourcegraph/README.md:
```diff
@@ -0,0 +1,73 @@
+## Further reading
+
+- [Use Datadog's Sourcegraph extension to navigate code and visualize service dependencies][4]
+Need help? Contact [Datadog support][3].
```
```suggestion

Need help? Contact [Datadog support][3].

```

kayayarai commented:
Hi @Joelkw I made one more formatting suggestion to add some whitespace around links. Also, the agent-integrations team needs to review the PR as well. My review was for the documentation only.

Joelkw commented:
Thanks @kayayarai ! I'll check on that tomorrow in my sync with some folks over at Datadog.

fanny-jiang commented:
Hi @Joelkw, thanks for opening this PR, it looks great! I've requested an integration logo design from our design team using the logo image that was provided in the PR. Could you also please add back the `metadata.csv` file? It is required per our [documentation](https://docs.datadoghq.com/developers/integrations/new_check_howto/?tab=configurationtemplate). If the integration doesn't collect any metrics, it can be an empty file with just the column headers. For example, https://github.com/DataDog/integrations-extras/blob/master/insightfinder/metadata.csv

fanny-jiang commented on sourcegraph/manifest.json:
```diff
@@ -0,0 +1,47 @@
+{
+  "integration_id": "sourcegraph-datadog-services-map-extension",
```
This is a bit long for an integration id. Would it be possible to change this to `sourcegraph` or something shorter? Keep in mind, this would be the unique identifying name for the integration.

Joelkw replied:
Got it! There may be other types of Sourcegraph integrations in the future so how about `sourcegraph-services-map`? Feel free to make that change if that's okay!

fanny-jiang replied:
`sourcegraph-service-map` sounds good!

Joelkw commented:
> Hi @Joelkw, thanks for opening this PR, it looks great! I've requested an integration logo design from our design team using the logo image that was provided in the PR. Could you also please add back the `metadata.csv` file? It is required per our [documentation](https://docs.datadoghq.com/developers/integrations/new_check_howto/?tab=configurationtemplate). If the integration doesn't collect any metrics, it can be an empty file with just the column headers. For example, https://github.com/DataDog/integrations-extras/blob/master/insightfinder/metadata.csv

Thanks @fanny-jiang ! Added back.

fanny-jiang commented on sourcegraph/metadata.csv:
```diff
@@ -0,0 +1 @@
+metric_name,metric_type,interval,unit_name,per_unit_name,description,orientation,integration,short_name
```
My apologies! I realized that you're using manifest v2 so you can remove this file.

fanny-jiang commented on .github/CODEOWNERS:
```diff
@@ -374,6 +375,9 @@
 /sortdb/*metadata.csv                    @namrata4 namrata.deshpande4@gmail.com @DataDog/documentation
 /sortdb/manifest.json                    @namrata4 namrata.deshpande4@gmail.com @DataDog/documentation
 /sortdb/README.md                        @namrata4 namrata.deshpande4@gmail.com @DataDog/documentation
+/sourcegraph/*metadata.csv               @sourcegraph/growth-and-integrations @DataDog/documentation
```
```suggestion
```

fanny-jiang commented on sourcegraph/manifest.json:
```diff
@@ -0,0 +1,47 @@
+  "app_id": "sourcegraph",
+  "type": "check",
+  "assets": {
+    "service_checks": {
+      "metadata_path": "assets/service_checks.json"
+    }
```
```suggestion
```

You can also remove this section along with the `assets/service_checks.json` file.

fanny-jiang commented:
Hi @Joelkw, have you had a chance to review my comments? We'd also like to confirm the `integration_id` that you'd like to go with since our team needs that for the logo design.

Joelkw commented on sourcegraph/manifest.json:
```diff
@@ -0,0 +1,47 @@
+{
+  "integration_id": "sourcegraph-datadog-services-map-extension",
```
```suggestion
  "integration_id": "sourcegraph-services-map",
```

Joelkw commented:
@fanny-jiang thanks for the ping! Just accepted all the comments and added the ID, thank you! Curious – is there another logo being designed in addition to the submitted logo? If so, can I ensure somehow that it will comply with the Sourcegraph logo guidelines: https://handbook.sourcegraph.com/departments/marketing/brand/brand_guidelines/logo/ ? Thank you!

fanny-jiang commented:
Hi @Joelkw, sorry for the delay! I passed along the included logo image from the PR to the design team as an example so the final logo should look pretty much like that. I'll also pass along the logo guidelines to the team to review.

Joelkw commented:
@fanny-jiang just checking back – is there anything further I need to do here to get this in?

fanny-jiang commented:
Hi @Joelkw, apologies for the delay. I tried deploying the changes to our staging environment to verify the tile. The manifest validations failed for a few new required fields:

```
Error validating manifest schema:
API input validation failed: {'classifier_tags': [u'Missing data for required field.'], 'app_uuid': [u'Missing data for required field.'], 'author': {'homepage': [u'Missing data for required field.'], 'name': [u'Missing data for required field.'], 'sales_email': [u'Missing data for required field.'], 'support_email': [u'Missing data for required field.']}}
1 invalid files
```
Can you please add these fields to the `manifest.json`? See our [docs](https://docs.datadoghq.com/developers/integrations/check_references/?tab=manifestversion2#manifest-file) for guidance.

You can generate a unique `app_uuid` by running the `ddev -e create <fake_check> -v2` scaffolding command, which will auto-generate a pre-populated manifest.json file. Feel free to discard the scaffolding changes after you obtain the app_uuid.

Joelkw commented:
Hello @yzhan289 , thanks for checking. I did, and I believe I just added what's needed. Does that work? :) 

If not, at this point, can I ask for help getting this over the finish line? Totally fine for anyone on the datadog side to make small tweaks to get this manifest to pass.

yzhan289 commented:
@Joelkw looks like there's a conflict with the `CODEOWNERS` file. I was trying to fix it myself and push it to your branch, but it kept getting rejected. This conflict is due to another PR adding an entry to `CODEOWNERS` in the same line as where you added yours. It should be easy to fix, and the result should look something like this:

```
...
/sortdb/                                 @namrata4 namrata.deshpande4@gmail.com
/sourcegraph/                            @sourcegraph/growth-and-integrations
/speedscale/                             @kenahrens ken@speedscale.com
/speedtest/                              @platinummonkey
...
```

Joelkw commented:
@yzhan289 I believe I fixed it? Does that work, and can we merge now? I added you to have write access to the repo, in case anything else needs to be adjusted (sorry about that, that's probably what blocked you before).

sarah-witt reviewed (commented):
Hi @Joelkw, the codeowners look good now, thanks! I also left one more comment. Finally, validations are failing on your pull request, but if you update your branch with master those should pass again.

FlorentClarret commented:
Hi @Joelkw, thanks for this PR. I fixed the last missing things for this PR but I can't push to your branch. Could you add me to your repository so we can merge this PR? Thanks!

yzhan289 commented:
To speed up the process for getting this integration merged, I've created a new branch off of this PR and applied the final change. PTAL: https://github.com/DataDog/integrations-extras/pull/1527

fanny-jiang commented:
Closing per https://github.com/DataDog/integrations-extras/pull/1527#discussion_r982704166
